src/main/java/org/greenbyme/angelhack/service/PostService.java

```diff
@@ -100,9 +102,13 @@ public PostSaveResponseDto savePosts(Long userId, PostSaveRequestDto requestDto,
     public Page<PostResponseDto> getPostsByMission(Long missionId, Pageable pageable) {
         Mission mission = missionRepository.findById(missionId)
                 .orElseThrow(() -> new MissionException(ErrorCode.INVALID_MISSION));
-        return personalMissionRepository.findAllByMissionId(mission.getId(), pageable)
+        List<PostResponseDto> res = personalMissionRepository.findAllByMissionId(mission.getId(), pageable)
                 .map(p -> postRepository.findByPersonalMissionId(p.getId()))
-                .map(post -> new PostResponseDto(post.getId(), post.getUser().getNickname(), post.getPicture(), post.getPostLikes().size()));
+                .stream()
+                .filter(Objects::nonNull)
+                .map(PostResponseDto::new)
+                .collect(Collectors.toList());
+        return new PageImpl<>(res, pageable, res.size());
```

Comment:
PostQueryDslImpl에서 
  
public Post findByPersonalMissionId(Long personalMissionId) {
}
의 반환값을 Optional로 설정하면, 

```suggestion
        List<PostResponseDto> res = personalMissionRepository.findAllByMissionId(mission.getId(), pageable)
                .stream()
                .map(p -> postRepository.findByPersonalMissionId(p.getId()))
                .filter(Optional::isPresent)
                .map(Optional::get)
                .map(PostResponseDto::new)
                .collect(Collectors.toList());
```
로 dto에 null 체크 로직을 뺄 수 있을거같아요